lit/docs/guides/pipeline-guides.lit

```diff
@@ -0,0 +1,662 @@
+    put steps. It's likely that a \reference{put-step} only needs a subset of
+    all available artifacts generated throughout the job.
+
+    There are two ways to specify which artifacts to send to a
+    \reference{put-step}. You can specify \code{detect} as the
+    \reference{schema.step.put-step.inputs} or you can pass in an exact list of
+    all artifacts the \reference{put-step} needs.
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: the-job
+      plan: # Get some artifacts
+      - get: apples
+      - get: apple-basket
+      - get: monkeys
+      # using detect
+      - put: apples-in-basket
+        input: detect
+        params:
+          apples-location: apples/location # matches the first get step
+          basket: apple-basket # matches the second get step
+      # or you can specify the exact inputs needed
+      - put: apples-in-basket
+        input: [apples, apple-basket]
+        params:
+          apples-location: apples/location
+          basket: apple-basket
+    }}
+
+  }
+
+  \section{
+    \title{Putting Task Config's in Files}
+
+    A lot of the pipeline examples that you will find on this site and in
+    resource repos will embed a \reference{schema.step.task-step.config} directly
+    in the pipeline. This is a nice way of clearly seeing what inputs/outputs the
+    task uses. Tasks are usually designed to be used in multiple places, maybe
+    with slightly different configuration. To support this scenario, most users
+    store task configs in files instead of embedding the config directly in the
+    pipeline.
+
+    Here's what this looks like in practice.
+
+    \titled-codeblock{todays-date.yml, stored in some git repo under a folder named "task"}{yaml}{{
+    platform: linux
+
+    image_resource: # define a default image for the task to use
+      type: registry-image
+      source:
+        repository: busybox
+
+    run:
+      path: date
+      args: ["+%Y-%m-%d"]
+    }}
+    Using the task in a pipeline.
+
+    \titled-codeblock{pipline.yml}{yaml}{{
+    resources:
+    - name: ci
+      type: git
+      source:
+        uri: http://github.com/username/my-ci-repo
+
+    jobs:
+    - name: the-best-job
+      plan:
+      - get: ci
+      - task: what-day-is-it
+        file: ci/tasks/todays-date.yml
+    }}
+  }
+
+  \section{
+    \title{\code{Get} Images for Tasks Instead of using Anonymous Image Resources}
+
+    It is easy to let Concourse fetch images for tasks right when they are
+    needed, it's the easy out-of-the-box solution. Another way is to pass the
+    image(s) for tasks as inputs to jobs.
+
+    \bold{Before - Anonymous Image Fetching for Tasks}
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: job
+      plan:
+      - task: simple-task
+        config:
+          platform: linux
+          image_resource: # anonymous image resource
+            type: registry-image
+            source: { repository: busybox }
```

Comment:
Weird it seems to remove the curly braces from the docs? :( Is it just my build that doesn't show it?

![Screen Shot 2021-05-18 at 3 09 50 PM](https://user-images.githubusercontent.com/19290703/118709558-22901e80-b7eb-11eb-93bf-b23ab7fa0227.png)

Maybe we should just move the `repository: busybox` onto a new line instead, so that it doesn't look like you can put it on the same line without the curly braces!

Reply:
yeah that's a weird thing, gotta do TRIPLE CURLY BRACES when declaring the codeblock to avoid this. Gonna put it on a newline instead.